generation/WinSDK/enums.json

```diff
@@ -34723,6 +34723,144 @@
       }
     ]
   },
+  {
+    "type": "uint",
+    "name": "DXGI_PRESENT",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_PRESENT_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGISwapChain",
+        "method": "Present",
+        "parameter": "Flags"
+      },
+      {
+        "interface": "IDXGISwapChain1",
+        "method": "Present1",
+        "parameter": "PresentFlags"
+      },
+      {
+        "interface": "IDXGIDecodeSwapChain",
+        "method": "PresentBuffer",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_ENUM_MODES",
```

Comment:
Yeah, seems really awkward. I'll leave this flag untouched until a decision is made.

Reply:
The docs call the set DXGI_ENUM_MODES. Stick with the documented name.

Reply:
@mikebattista how about the other fields that @riverar asked me to add `_FLAGS` to, despite not being documented like that?

Reply:
If there's a documented name already we should use that.

Reply:
I've reverted the `_FLAGS` suggestion from @riverar for two enums that already have upstream documentation:

- `DXGI_PRESENT`
- `DXGI_SHARED_RESOURCE_RW`

I've left the others like this, and added a documentation URL to a function that describes their values (unsure if this is desired?):

- `DXGI_CREATE_FACTORY_FLAGS`
- `DXGI_MAP_FLAGS`
- `DXGI_MWA_FLAGS`
- `DXGI_RESOURCE_PRIORITY` (no rename was suggested here)

Reply:
@riverar any comments here? This looks to be the last issue to resolve (thus far), and I'd love to get this change in before the next `win32metadata` release to reap the benefits together with the Agility SDK 613 regen 😄

Reply:
Checking those last four now, one moment.

Reply:
tl;dr: LGTM. If any of the "no existing enumeration" assertions below are invalid, please let me know!

Re: `DXGI_CREATE_FACTORY_*`, these are documented as _flags_ to be passed into CreateDXGIFactoryN functions. Currently its only value is: `#define DXGI_CREATE_FACTORY_DEBUG 0x1`. There is no existing enumeration that captures these constants, therefore I propose: `DXGI_CREATE_FACTORY_FLAGS`.

Re: `DXGI_MAP_*`, these are documented as _flags_ to be passed into IDXGISurface::Map and possibly others. It has values such as DXGI_MAP_READ, [...]WRITE, and [...]DISCARD. There is no existing enumeration that captures these constants, therefore I propose: `DXGI_MAP_FLAGS`. This would also align with how we handle similar flags, such as `REG_SAM_FLAGS`.

Re: `DXGI_MWA_*`, these are documented as _flags_ to be passed into IDXGIFactory::MakeWindowAssociation. It has values such as DXGI_MWA_NO_ALT_ENTER. There is no existing enumeration that captures these constants, therefore I propose: `DXGI_MWA_FLAGS`.

Re: `DXGI_RESOURCE_PRIORITY`, these are documented as _mutually exclusive constants_ to be passed into IDXGIResource::SetEvictionPriority. It has values such as DXGI_RESOURCE_PRIORITY_LOW. There is no existing enumeration that captures these constants, therefore I propose keeping the original name: `DXGI_RESOURCE_PRIORITY`.

Reply:
Thanks for the detailed clarification, that makes total sense!

Let's mark this thread as resolved.